### DISCUSSION TITLE
findUnique with more than one arg

### DISCUSSION THREAD

**Discussion by ahmdtalat** [prisma/prisma#5405] [Q&A] [answered]
Hello Prisma, I ran into an issue where I need to check if a user exists by his username and email since both are unique fields in the database, but I got an error. 
`Argument where of type UserWhereUniqueInput needs exactly one argument, but you provided username and email. Please choose one.`
so, is there a way to execute this query just once?. instead of running one for each like the following

```js
const user = await prisma.user.findUnique({
        where: {
          username,
          email,
        },
      });
```
and not like this 
```js
const user = await prisma.user.findUnique({
        where: {
          username,
        },
      });

const user = await prisma.user.findUnique({
        where: {
          email,
        },
      });

```

**Comment by ryands17** [comment, marked as answer]
@ahmdtalat 👋 

In this case, you need to use [findFirst](https://www.prisma.io/docs/concepts/components/prisma-client/crud#findfirst). So your query would be:

```ts
const user = await prisma.user.findFirst({
        where: {
          username,
          email,
        },
});
```

**Reply by williamluke4** [reply]
Building on this you can also use `@@unique` which will allow you to use `findUnique` with both `username` and `email`. Please see the  [docs](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#unique-1)

**Reply by ahmdtalat** [reply]
> @ahmdtalat 
> 
> In this case, you need to use [findFirst](https://www.prisma.io/docs/concepts/components/prisma-client/crud#findfirst). So your query would be:
> 
> ```ts
> const user = await prisma.user.findFirst({
>         where: {
>           username,
>           email,
>         },
> });
> ```
Heyyy, does FindFrist work as (username OR email) or (username AND email)?

**Reply by ahmdtalat** [reply]
Hi @williamluke4, I just used $queryRaw instead, no idea which one is more efficient. but it worked just fine.thnx

**Reply by ryands17** [reply]
It will work as `AND`. If you want `OR`. you would need to add `OR` inside where:

```ts
const user = await prisma.user.findFirst({
        where: {
          OR: [{ username }, { email }],
        },
});
```

**Reply by ahmdtalat** [reply]
@ryands17 thanks a lot, that is more elegant than a SQL query :rocket:

**Reply by HkSolDev** [reply]
@ryands17 thanks this work as I want to check for the both username and email

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
